hmcts/pre-api
# Audit a DELETE activity when a delete controller is called on an entity

### JIRA ticket(s)

- S28-3081

### Change description

- Replace all 'manual' updates bypassing the callbacks in JPA with typical updates and set flag so the `AuditListener` knows if it is a delete occurring


> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**

- After creating some entities which we audit (users, recordings, bookings) delete them and observe changes in the audit table. Deletes should be visible whereas before these were recorded as Updates 
- NOTE: Hibernate / JPA has a lot of abstracted magic which appears to do things like UPDATE after a CREATE so you may see this in the audit. You should, however, only see 1 delete happen.
